bpf/bpf_sock.c

```diff
@@ -452,10 +454,19 @@ __sock4_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 __section("cgroup/connect4")
 int cil_sock4_connect(struct bpf_sock_addr *ctx)
 {
+	int err;
+
 	if (sock_is_health_check(ctx))
 		return __sock4_health_fwd(ctx);
 
-	__sock4_xlate_fwd(ctx, ctx, false);
+	err = __sock4_xlate_fwd(ctx, ctx, false);
+	if (err == -EHOSTUNREACH || err == -ENOMEM) {
+#ifdef HAVE_SET_RETVAL
+		bpf_set_retval(err);
+#endif
```

Comment:
tiny nit, but not blocking: I'd make a small static inline helper like `set_retval(err)` and just hide the ifdef there, so it doesn't need to be on all callsites.